### PR TITLE
Update to Chef 14. Update OS version support for Ubuntu and OSX

### PR DESCRIPTION
**Pull request by rwinfosec**
Bumped to Chef 14.
Updated and fixed support for Ubuntu 16.04 and 18.04
Updated support for OSX to 10.13
fixed formatting: https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/FrozenStringLiteralComment

### CODE REVIEW

File: test/integration/default/inspec/osquery_spec.rb [hunk 1]
```diff
@@ -27,6 +14,29 @@
...
+      'logger_plugin' => 'filesystem',
+      'schedule_splay_percent' => 10,
+      'events_expiry' => 3600,
+      'verbose' => false,
+      'worker_threads' => 2,
+      'logger_path' => '/var/log/osquery'
+    }
+
+    expected_schedule = {
+      'info' => { 'query' => 'SELECT * FROM osquery_info;', 'interval' => 86_400 }
```
Comment (charlesdunbar) [inline test/integration/default/inspec/osquery_spec.rb:30]: typo on the interval

File: test/integration/default/inspec/osquery_spec.rb [hunk 2]
```diff
@@ -65,4 +66,27 @@
...
```
Comment (charlesdunbar) [inline test/integration/default/inspec/osquery_spec.rb:83]: Same typo with `interval`

**Review by charlesdunbar — CHANGES_REQUESTED** [review]

File: test/integration/default/inspec/osquery_spec.rb (hunk 1 above)
Comment (rwinfosec) [reply test/integration/default/inspec/osquery_spec.rb:30]: Updated!

File: test/integration/default/inspec/osquery_spec.rb (hunk 2 above)
Comment (rwinfosec) [reply test/integration/default/inspec/osquery_spec.rb:83]: Updated!

Comment (charlesdunbar) [reply test/integration/default/inspec/osquery_spec.rb:83]: TIL ruby can just add `_` anywhere in a number for readability, and rubocop demands it for thousands separation.

**Review by charlesdunbar — APPROVED** [review]

**Pull request merged**